src/services/TemplatingService.ts

```diff
@@ -0,0 +1,150 @@
+import { messaging as firebaseMessaging } from 'firebase-admin';
+import Mail from '@/interfaces/Mail';
+import Template from '../interfaces/Template';
+import TemplatePayload from '@/interfaces/TemplatePayload';
+import UserRessource from '@/interfaces/UserRessource';
+import Utils from '../utils';
+
+const MAIL_MESSAGE = 'MAIL';
+const PUSH_MESSAGE = 'PUSH';
+
+export default class TemplatingService {
+  // region public static methods
+  // endregion
+
+  // region private static methods
+
+  private static insertMessagePayload(template: any, payload: any, platformId: string): any {
+    const config = Utils.getPlatformConfig(platformId);
+    return {
+      from: config.mail.defaults.from,
+      data: template.data,
+    };
+  }
+
+  private static insertLanguagePayload(template: any, payload: any): any {
+    // TODO: Insert message content into template
+    return Object.assign(template, {
```

Comment:
What is the benefit of Object.assign here, why don't you just return a new object like `return { ... }`?

Reply:
This function should extend the given `template` object. I use `Object.assign()` two merge the `template` object with the new properties.

Reply:
Thank you :)